include/kokkos-utils/callbacks/TimerListenerBeginEnd.hpp

```diff
@@ -0,0 +1,74 @@
+#ifndef KOKKOS_UTILS_CALLBACKS_TIMERLISTENERBEGINEND_HPP
+#define KOKKOS_UTILS_CALLBACKS_TIMERLISTENERBEGINEND_HPP
+
+#include "kokkos-utils/callbacks/EventBeginEndEventIdMatcher.hpp"
+#include "kokkos-utils/callbacks/EventOnExecMatcher.hpp"
+#include "kokkos-utils/callbacks/TimerListener.hpp"
+#include "kokkos-utils/timer/Duration.hpp"
+#include "kokkos-utils/timer/Timer.hpp"
+
+namespace Kokkos::utils::callbacks
+{
+
+//! Timer for events that are enqueued on @ref exec.
+template <Kokkos::utils::concepts::ExecutionSpace Exec>
+struct EnqueuedEventTimer
```

Comment:
We can probably delete the enqueuedeventimer because it's covered by the other matcher.

Reply:
Let's try.

Reply:
It's actually not possible, because neither `TimerListener` nor our `Kokkos::utils::timer::Timer` store the `exec` instance.

Reply:
We kept it. Because it paves the way to become a bit more general later.